crates/krun/src/bin/krun.rs

```diff
@@ -208,8 +203,26 @@ fn main() -> Result<()> {
         }
     }
 
-    let krun_guest_path = find_krun_exec("krun-guest")?;
-    let krun_server_path = find_krun_exec("krun-server")?;
+    let krun_guest_path = {
+        const PROGRAM: &str = "krun-guest";
+        let path = find_krun_exec(PROGRAM)?;
+        CString::new(path.to_str().with_context(|| {
+            format!("Failed to process {PROGRAM:?} path as it contains invalid UTF-8")
+        })?)
+        .with_context(|| {
+            format!("Failed to process {PROGRAM:?} path as it contains NUL character")
+        })?
+    };
+    let krun_server_path = {
+        const PROGRAM: &str = "krun-server";
+        let path = find_krun_exec(PROGRAM)?;
+        CString::new(path.to_str().with_context(|| {
```

Comment:
Why can't we do the `CString` conversion in `find_krun_exec`.

Reply:
Sure we could, but we don't always want a `CString`. Returning a `PathBuf` for a path is less surprising / more flexible?

Reply:
The two uses we have right now need a `CString`. I think it's fine to keep it as it, and generalize it if/when we need to use it in more places.

Reply:
Alright.